shoop/xtheme/plugins/widgets.py

```diff
@@ -90,3 +91,38 @@ def decompress(self, value):
         else:
             out_dict[FALLBACK_LANGUAGE_CODE] = value
         return out_dict
+
+
+class XThemeModelChoiceWidget(forms.Select):
+    def render(self, name, value, attrs=None, choices=()):
+        return mark_safe(
+            render_to_string("shoop/xtheme/_model_widget.jinja", {
+                "name": name,
+                "selected_object": [object for pk, object in self.choices if pk == int(value)][0] if value else None,
```

Comment:
... Oh, wait, you're only using this to render `selected`?

Why not just pass `value` down to the template context and do `if value == object.pk` in the render loop there?